internal/cmd/providers.go

```diff
@@ -65,16 +66,17 @@ func (o providerAPIOptions) Validate(providerKind string) error {
 
 type providerEditOptions struct {
 	ClientSecret       string
+	SCIM               bool
 	ProviderAPIOptions providerAPIOptions
 }
 
 func (o providerEditOptions) Validate(providerKind string) error {
-	if o.ClientSecret == "" && o.ProviderAPIOptions.PrivateKey == "" && o.ProviderAPIOptions.ClientEmail == "" && o.ProviderAPIOptions.WorkspaceDomainAdminEmail == "" {
-		return fmt.Errorf("Please specify a field to update.'\n\n%s", newProvidersEditCmd(nil).UsageString())
+	if !o.SCIM && o.ClientSecret == "" {
+		return fmt.Errorf("scim or client-secret flag are required for updates")
 	}
 
-	if providerKind != "google" && o.ClientSecret == "" {
-		return fmt.Errorf("Client secret flag must be specified when updating an identity provider that isn't of kind Google")
+	if o.ClientSecret == "" && o.ProviderAPIOptions.PrivateKey == "" && o.ProviderAPIOptions.ClientEmail == "" && o.ProviderAPIOptions.WorkspaceDomainAdminEmail == "" {
+		return fmt.Errorf("client secret, private key, client email, and workspace domain admin email are required.'\n\n%s", newProvidersEditCmd(nil).UsageString())
```

Comment:
Should the condition be `||` instead of `&&` ? The error message reads like all of the values are required, but the condition is "one of" them is required.

I wonder if we should do the `newProvidersEditCmd(nil).UsageString()` printing from the caller. I guess any validation of options would benefit from the usage text.

Reply:
This edit command validation was wrong, fixed it